### DISCUSSION TITLE
Defining boundary condition for a set of points (PointSetBC)

### DISCUSSION THREAD

**Discussion by DeepaMahm** [lululxvi/deepxde#1569] [Q&A]
Dear Prof. @lululxvi ,

I would like to know how to define boundary conditions for a set of points (2D geometry, with mesh shown below).

![image](https://github.com/lululxvi/deepxde/assets/29662579/39c51772-1681-4169-8a42-c7d961e89d82)

For instance, for  a 1D geometry, to specify Dirichlet boundary condition on the left I use (1D diffusion problem)
 

```
def boundary_l(x, on_boundary):
    """
    :param x:
    :param on_boundary: all boundary points
    :return:
    """
    return on_boundary and dde.utils.isclose(x[0], 0)  # définit le boundary x=0, left

```
```
    geom = dde.geometry.Interval(0, 1)
    timedomain = dde.geometry.TimeDomain(0, 1)
    geomtime = dde.geometry.GeometryXTime(geom, timedomain)

    bc_l = dde.icbc.DirichletBC(geomtime, lambda x: 5, boundary_l)
    bc_r = dde.icbc.NeumannBC(geom, lambda x: 0, boundary_r)

```

Similarly, I am not sure how to specify Dirichlet (left)/Neumann (right ) boundary condition using `DirichletBC` function.

Suggestions on how to specify Dirichlet (left)/Neumann (right ) BC for a set of points  (to solve forward problem) will be really helpful.

**Comment by DeepaMahm** [comment]
Prof. @lululxvi ,

Could you please have a look into this when it is possible for you?
I think I could use `PointSetBC` to specify the Dirichlet boundary condition for a set of points. However I am not sure how Neumann boundary conditions can be specified for a set of points. 

Could you please give an example?

**Reply by lululxvi** [reply]
Yes, use `PointSetBC` for Dirichlet BC.

For Neumann BC, you could implement new code based on `PointSetBC` and `NeumannBC`.
